blip-00xx.md

```diff
@@ -0,0 +1,220 @@
+
+The proposal is to add the following new [tagged field][tagged-fields] to the 
+set defined in [BOLT 11][bolt11]:
+
+- `b` (20): `data_length` variable. One or more entries each containing a 
+   blinded payment path for a private route; there may be more than one `b` 
+   field. The field uses the `blinded_payinfo` type described below which draws 
+   heavily on the proposed encoding of the `blinded_payinfo` subtype defined in 
+   the [Offers proposal][offers].
+
+1. subtype: `blinded_payinfo`
+2. data:
+  * [`u32`:`fee_base_msat`]
+  * [`u32`:`fee_proportional_millionths`]
+  * [`u16`:`cltv_expiry_delta`]
+  * [`u64`:`htlc_minimum_msat`]
+  * [`u64`:`htlc_maximum_msat`]
+  * [`u16`:`flen`]
+  * [`flen*byte`:`features`]
+  * [`33*byte`:`first_ephemeral_blinding_point`]
+  * [`byte`:`num_hops`]
+  * [`num_hops*blinded_hop`:`blinded_hops`]
+
+1. subtype: `blinded_hop`
+2. data:
+   * [`33*byte`:`blinded_node_pubkey`]
+   * [`2*byte`:`cipher_text_length`]
+   * [`cipher_text_length*byte`:`cipher_text`]
+
+The `blinded_node_pubkey` of the first `blinded_hop` in a `blinded_payinfo` is
+the real public key of the blinded path's introduction node. 
+
+### Requirements
+
+An invoice containing the `b` field type:
+- MUST not contain the `r` field type.
+- MUST not contain the `s` field type since a payment address in the context of
+  blinded payments does not make sense since the recipient is able to use the
+  `path_id` in the `encrypted_recipient_data` for the same purpose.
+- SHOULD sign the invoice with a private key that is not the same as their 
+  public node ID and should not set the destination node (`n` field).
+
+### Size Restrictions
+
+#### `data_length` Limit
+
+In order to conform to any existing BOLT 11 invoice parser, each new tagged 
+field must use the `data_length` encoding defined there. This means that the 
+maximum size of any _single_ encoded blinded path is 639 bytes. 
+
+What follows is a rough estimation of the maximum number of hops we can include 
+in a single blinded path. It assumes that each hop's cipher text is the same 
+length. 
+
+##### Cipher Text Size Estimation
+
+First, a rough estimation of the average cipher text length is required. A 
+forwarding node in a blinded path will receive a cipher text payload containing
+the following data:
+
+- `padding`: optional
+- `short_channel_id` of 8 bytes
+- `payment_relay`: 
+    * 2 byte `cltv_expiry_delta`
+    * 4 byte `fee_proportional_millionths`
+    * 4 byte `fee_base_msat`
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
+    * 8 byte `htlc_minimum_msat`
+- `allowed_features`: optional
+
+If we assume that the `allowed_features` vector is not set, then this comes to 
+a total of 30 mandatory bytes. 
+
+For the recipient node, it will receive a cipher text payload containing:
+
+- `padding`: optional
+- `path_id`: let's assume that this is 32 bytes like the existing payment 
+   address.
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
```

Comment:
still unsure why the receiver needs the payment_contraint data, it created the invoice in the first place ?

Reply:
so i think the idea here is so that the recipient doesnt need to go store this expiry as a separate thing. It basically just sends this message to itself. Got the idea from the example used in the [proposal doc](https://github.com/lightning/bolts/blob/master/proposals/route-blinding.md#blinded-payments)

Reply:
defs an open discussion though. Noted in the missing pieces part here: https://github.com/lightningnetwork/lnd/pull/8752#issuecomment-2189971927